java/core/src/main/java/co/worklytics/psoxy/rules/salesforce/PrebuiltSanitizerRules.java

```diff
@@ -245,10 +310,105 @@ private static JsonSchemaFilterUtils.JsonSchemaFilter jsonSchemaForQueryResult(J
                             .build());
                     put("records", JsonSchemaFilterUtils.JsonSchemaFilter.builder()
                             .type("array")
-                            .items(recordSchema)
+                            .items(JsonSchemaFilterUtils.JsonSchemaFilter.builder()
+                                    .type("object")
+                                    .properties(Collections.emptyMap())
+                                    ._if(JsonSchemaFilterUtils.ConditionJsonSchema.builder().properties(new LinkedHashMap<String, JsonSchemaFilterUtils.JsonSchemaFilter>() {{ //req for java8-backwards compatibility
+                                        put("attributes", buildAttributeWithType("Account"));
+                                    }}).build())
+                                    ._then(JsonSchemaFilterUtils.ThenJsonSchema.builder().properties(ACTIVITY_HISTORIES_SUB_QUERY_RESULT_PROPERTY_SCHEMA).build())
+                                    .build())
                             .build());
                 }})
                 .build();
     }
-}
 
+    private static JsonSchemaFilterUtils.JsonSchemaFilter jsonSchemaForQueryResult() {
+        return JsonSchemaFilterUtils.JsonSchemaFilter.builder()
+                .type("object")
+                // Using LinkedHashMap to keep the order to support same
+                // YAML serialization result
+                .properties(new LinkedHashMap<String, JsonSchemaFilterUtils.JsonSchemaFilter>() {{ //req for java8-backwards compatibility
+
+                    put("totalSize", JsonSchemaFilterUtils.JsonSchemaFilter.builder()
+                            .type("integer")
+                            .build());
+                    put("done", JsonSchemaFilterUtils.JsonSchemaFilter.builder()
+                            .type("boolean")
+                            .build());
+                    put("nextRecordsUrl", JsonSchemaFilterUtils.JsonSchemaFilter.builder()
+                            .type("string")
+                            .build());
+                    put("records", JsonSchemaFilterUtils.JsonSchemaFilter.builder()
+                            .type("array")
+                            .items(JsonSchemaFilterUtils.JsonSchemaFilter.builder()
+                                    .oneOf(Arrays.asList(
```

Comment:
This is the model for query pagination response, something like:

```
oneoff : [
{
if: is user
then: get the user response model
},
{
if: is account/activityhistory
then: get the account/activityhistory response model
}]
```
 
In this case, account/activityhistory have merged rules as the queries are:
- accounts: get all the accounts
- activity history: get the accounts with activity history, as an inner query